openrpc.yaml

```diff
@@ -841,6 +841,76 @@ methods:
         result:
           name: eth_unsubscribeExampleResult
           value: true
+  - name: wallet_swapAsset
+    tags:
+      - $ref: '#/components/tags/MetaMask'
+      - $ref: '#/components/tags/Experimental'
+    summary: Allows dApps to request MetaMask to perform a token swap operation.
+    description: >-
+      This method enables dApps to initiate a token swap directly within MetaMask, improving the user experience by streamlining the process of swapping tokens. The method accepts parameters for the source and destination tokens and initiates a token swap operation.
+    params:
+      - name: SwapAssetParameter
+        required: true
+        schema:
+          title: SwapAssetParameter
+          type: object
+          required:
+            - fromToken
+            - toToken
+            - userAddress
+          properties:
+            fromToken:
+              type: array
+              items:
+                type: object
+                required:
+                  - address
+                properties:
+                  address:
+                    type: string
+                    description: The CAIP-10 formatted address of the source token.
+                    $ref: '#/components/schemas/Caip10Address'
+                  value:
+                    type: string
+                    description: (Optional) The amount of the source token to be swapped, in wei, as a hexadecimal string, defaults to 0.
+                    nullable: true
+            toToken:
+              type: object
+              required:
+                - address
+              properties:
+                address:
+                  type: string
+                  description: The CAIP-10 formatted address of the destination token.
+                  $ref: '#/components/schemas/Caip10Address'
+            userAddress:
+              type: string
+              description: The CAIP-10 formatted address of the user performing the swap.
+              $ref: '#/components/schemas/Caip10Address'
+    result:
+      name: SwapAssetResult
+      description: A boolean indicating if the swap was initiated successfully.
+      schema:
+        type: boolean
+    errors:
+      - code: -36000
+        message: "Swap is not available on this chain {{chainName}}"
+      - code: -32004
```

Comment:
```suggestion
      - code: 6001
```

Reply:
I think [invalid request](https://github.com/MetaMask/rpc-errors/blob/ac963b4f3070b32c27d7656cc51fd4f272a751db/src/error-constants.ts#L10C5-L10C28) is more appropriate for this error